error: no rules expected the token `:`
 --> $DIR/invalid-bounds.rs:4:25
  |
4 |     struct A<T: 'static : ?Sized> { //~ ERROR no rules expected the token `:`
  |                         ^ no rules expected this token in macro call

error: expected one of `+`, `,`, `=`, or `>`, found `:`
  --> $DIR/invalid-bounds.rs:9:1
   |
9  | / pin_project! {
10 | |     struct B<T: Sized : 'static> { //~ ERROR expected one of `+`, `,`, `=`, or `>`, found `:`
11 | |         field: T,
12 | |     }
13 | | }
   | | ^
   | | |
   | |_expected one of `+`, `,`, `=`, or `>`
   |   in this macro invocation
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
